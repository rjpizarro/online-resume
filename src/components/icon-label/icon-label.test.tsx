import React from 'react'
import renderer from 'react-test-renderer'
import { render } from "@testing-library/react"

import { FaGamepad } from "react-icons/fa"
import IconLabel from './index'

describe("Icon Label", () => {
  it("render normally with required props", () => {
    const tree = renderer
      .create(<IconLabel icon={<FaGamepad />} label="label" />)
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  it("pass iconColor prop to an icon from react-icons library", () => {
    const { getByTestId } = render(
      <IconLabel
        icon={<FaGamepad  data-testid="icon-test" />}
        label="label"
        iconColor="#FFF"
      />
    )

    expect(getByTestId("icon-test")).toHaveProperty("style.color", 'rgb(255, 255, 255)')
  })
})