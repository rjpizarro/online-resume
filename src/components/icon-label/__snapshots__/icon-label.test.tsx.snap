// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon Label render normally with required props 1`] = `
<div
  className="makeStyles-container-2"
>
  <div
    className="makeStyles-iconWrapper-3"
  >
    <svg
      fill="currentColor"
      height="40px"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 640 512"
      width="40px"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M480 96H160C71.6 96 0 167.6 0 256s71.6 160 160 160c44.8 0 85.2-18.4 114.2-48h91.5c29 29.6 69.5 48 114.2 48 88.4 0 160-71.6 160-160S568.4 96 480 96zM256 276c0 6.6-5.4 12-12 12h-52v52c0 6.6-5.4 12-12 12h-40c-6.6 0-12-5.4-12-12v-52H76c-6.6 0-12-5.4-12-12v-40c0-6.6 5.4-12 12-12h52v-52c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v52h52c6.6 0 12 5.4 12 12v40zm184 68c-26.5 0-48-21.5-48-48s21.5-48 48-48 48 21.5 48 48-21.5 48-48 48zm80-80c-26.5 0-48-21.5-48-48s21.5-48 48-48 48 21.5 48 48-21.5 48-48 48z"
      />
    </svg>
  </div>
  <h6
    className="MuiTypography-root makeStyles-label-4 MuiTypography-subtitle2"
  >
    label
  </h6>
</div>
`;
